src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4126,16 +4126,27 @@ trait Types
       val extrapolation = new ExistentialExtrapolation(tparams)
       if (flipVariance) extrapolation.variance = Contravariant
       val tpe1     = extrapolation extrapolate tpe
```

Comment:
I just set a breakpoint here in the debugger to see what sort of data arrives.

I noticed this is called with `NoType`, we could add a fast path for that.

It would also be worthwhile adding a fast path for `tparams.size == 1`, which is pretty common and doesn't need the `ListBuffer` allocations for the general-arity algorithm.

Reply:
I have added a shortcut for both cases.